### DISCUSSION TITLE
Lots of Spam issues

### DISCUSSION THREAD

**Discussion by Softbox44** [docker-mailserver/docker-mailserver#4008] [Q&A]
I keep getting a lot of spam email. They are product offerings which could be real but they are just repeated unsolicited mail-shots. How can I reject those? Can I train the spam-filter? Can I **manually** add a domain for rejection?

These are my settings:
```
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_CLAMAV=0 
      - ENABLE_FAIL2BAN=1
      - ENABLE_POSTGREY=0
```

**Comment by glime-ai** [comment]
**Please note** that this response was created using generative AI, leveraging previously available information from the repository. This information may be helpful. However, it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To address the issue of receiving a lot of spam emails, you can take several steps to enhance the spam filtering capabilities of your Docker Mailserver setup. Here are some recommendations based on your current settings:

1. **Train SpamAssassin**: You can improve SpamAssassin's ability to recognize spam by training it with examples of spam and non-spam (ham) emails. This can be done using the `sa-learn` command. You can set up a cron job to automate this process. For example:
   ```bash
   # Learn spam every day at 2:00 AM
   0 2 * * * docker exec mailserver sa-learn --spam /var/mail/example.com/username/.Junk --dbpath /var/mail-state/lib-amavis/.spamassassin
   # Learn ham at different intervals
   15 2 * * * docker exec mailserver sa-learn --ham /var/mail/example.com/username/.Archive* --dbpath /var/mail-state/lib-amavis/.spamassassin
   ```
   Ensure you have a `/var/mail-state` volume mounted for this to work effectively.

2. **Manually Add Domain for Rejection**: You can manually block domains by adding them to the SpamAssassin configuration. Edit the `local.cf` file or create custom rules in `spamassassin-rules.cf` to specify domains that should be blocked. For example:
   ```conf
   blacklist_from *@unwanteddomain.com
   ```
   This rule will mark all emails from `unwanteddomain.com` as spam.

3. **Adjust SpamAssassin Settings**: Consider adjusting the thresholds for tagging and blocking spam. For example, you can change the `SA_TAG2` and `SA_KILL` levels to more aggressive values if too much spam is getting through.

4. **Enable ClamAV**: Although you currently have ClamAV disabled (`ENABLE_CLAMAV=0`), enabling it can help in filtering out emails with malicious attachments, which are often spam. Set `ENABLE_CLAMAV=1` to activate this feature.

5. **Consider Additional Anti-Spam Measures**: If SpamAssassin and ClamAV are not sufficient, consider enabling additional anti-spam measures like Rspamd, which can work alongside or replace SpamAssassin and provides advanced filtering capabilities.

6. **Review and Optimize Configuration**: Regularly review your mail server's logs and configurations to optimize the spam filtering rules based on the types of spam emails you receive.

By implementing these steps, you should see a reduction in the amount of spam reaching your inbox. Training SpamAssassin regularly and fine-tuning your spam filtering rules will be key to maintaining effective spam protection.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=0d5b86fb7eea4a54acac405212c29aba&utm_medium=github).

**Comment by polarathene** [comment]
AFAIK there is a way for SA to train, look over our docs like the FAQ page and through issues history here.

You could also consider trying rspamd as per our docs guidance to see if that does a better job. We intend to make rspamd default in future, usually it's a better experience but we want to make sure that it's integrated well before that happens, so if you try it please let us know if it was better/worse 👍
